File: php/features.php
<?php
declare(strict_types=1);

// BluefinPayconex SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class BluefinPayconexFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BluefinPayconexBaseFeature();
            case "test":
                return new BluefinPayconexTestFeature();
            default:
                return new BluefinPayconexBaseFeature();
        }
    }
}

File: py/features.py
# BluefinPayconex SDK feature factory

from feature.base_feature import BluefinPayconexBaseFeature
from feature.test_feature import BluefinPayconexTestFeature


def _make_feature(name):
    features = {
        "base": lambda: BluefinPayconexBaseFeature(),
        "test": lambda: BluefinPayconexTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
